src/windows/cpu.rs

```diff
@@ -147,15 +147,15 @@ impl Drop for InternalQuery {
     fn drop(&mut self) {
         unsafe {
             for (_, counter) in self.data.iter() {
-                PdhRemoveCounter(counter.0);
+                PdhRemoveCounter(*counter);
             }
 
             if !self.event.is_invalid() {
                 let _err = CloseHandle(self.event);
             }
 
-            if !self.query.is_invalid() {
-                PdhCloseQuery(self.query.0);
+            if !(self.query == -1 || self.query == 0) {
```

Comment:
Yeah it's not great, it just duplicates the `is_invalid` code but I wasn't sure even /that/ was actually correct in this case. Usually these windows functions are already hardened against invalid handles so it might just be OK to always call `PdhCloseQuery` instead of checking on our side.

Ideally there'd be a `PDH_HQUERY` as used by the Pdh* functions but no such type exists in windows-rs.

Reply:
I opened https://github.com/microsoft/windows-rs/issues/3200 for this. I'm really not sure what to do here... Is there any internals improvements in `windows` crate worth merging this PR as is even with the new code?

If not, would you be open to create a newtype which would wrap this value so we at least know what we're manipulating?

Reply:
> I opened https://github.com/microsoft/windows-rs/issues/3200 for this. I'm really not sure what to do here... Is there any internals improvements in `windows` crate worth merging this PR as is even with the new code?
> 
> If not, would you be open to create a newtype which would wrap this value so we at least know what we're manipulating?

It would be nice if this were merged so that my dependency tree doesn't use 4 or 5 different versions of windows-rs. :)

Reply:
It was merged. Just need to wait for `windows-rs` 0.59 version to be released now I suppose. :)

Reply:
Wouldn't it be a better idea to merge this and make the code look like 0.59 in the meantime. windows-rs upgrades are infrequent and (by us) kind of unwanted too since they're a massive ecosystem wide upgrade that needs to happen.

Reply:
Not really. The code is much less good with the 0.58 version and I'd prefer to avoid having "bad code" until next windows-rs release.

Reply:
> It was merged. Just need to wait for `windows-rs` 0.59 version to be released now I suppose. :)

I'm confused. You say it was merged, but the PR is still open.

Reply:
The issue I opened was merged in https://github.com/microsoft/win32metadata/commit/ff991cf3d93037d4429f124684b9a22f7acb6a7b. Now I'm waiting for windows-rs to update their dependency.

> (why is it not called "main"?)

It was created with the branch named `master` so don't see a reason to change it.